CHANGELOG.md

```diff
@@ -54,6 +54,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `Interval` class now supports degenerate intervals containing only one element
 - `add_fake_results` now directly processes `Target` objects instead of a `Campaign`
 - `path` argument in plotting utility is now optional and defaults to `Path(".")`
+- `UnusedObjectWarning` by non-predictive recommenders is now ignored during simulations
```

Comment:
Hi @Scienfitz, perhaps this is bullshit, but I just thought about whether this is actually the best angle from which to solve the problem. At its heart, I think the problem is not related to simulation but kicks in already on level earlier, namely: conceptually, it already doesn't make much sense to even create a campaign with the intention of using non-predictive recommenders, I think. Because already in the second iteration, you'll automatically run into the problem, while in the first iteration, you could just call the recommender directly (because there is no metadata handling required in the first place). So I'm wondering if it's more clever to just trigger the warning when creating a campaign with a "useless" configuration?

Reply:
i think thats too strict, i recall you making examples of for instance sequential recommenders that throw in a clustering every now and then (we dont know whether that would be a great idea but it could also be), this would trigger the warning here too
in that sense it is an issue with simulation passing the objective and measurement blindly without further consideration, the warning filter fixes this

Reply:
Comment upfront: We can absolutely keep the current version if it brings us faster to the goal for the hackathon. I'm just continuing the discussion because I feel that the underlying issue is not yet fully resolved.

To your point: I think I'm not yet convinced. Even when you say you run something like a clustering in between (btw: can't remember suggesting that but perhaps we are talking about different things), then again this can be separate into the two cases:
* Either, the clustering is data-based. In that case, everything is fine and no exception needs to be thrown at all
* Or, the clustering ignores the data collected up to that point. But then it makes no sense to run the clustering in the middle of a campaign and we could have shown the warning once, immediately when creating the campaign where we specified that a non-data clustering will be executed at a point after the first iteration

Reply:
I suggest you add this to the excel sheet that tracks the things that need to be valdiated /warned

until all of this is implemented (and it is certainly not the aim of this PR), we need a fix for the warning bombardement that is currently unleashed to users of simulation

Reply:
Sounds good to me, have added it. Consider it closed for this PR